.taskcluster.yml

```diff
@@ -128,78 +123,97 @@ tasks:
+          then:
+            $let:
+              pull_request_title: ${event.pull_request.title}
+              pull_request_number: ${event.pull_request.number}
+              pull_request_url: ${event.pull_request.html_url}
+            in:
               $mergeDeep:
                 - {$eval: 'default_task_definition'}
                 - scopes:
-                    - ${assume_scope_prefix}:branch:${short_head_branch}
+                    - ${assume_scope_prefix}:pull-request
                   payload:
                     command:
                       - >-
                         git fetch ${repository} ${head_branch}
                         && git config advice.detachedHead false
                         && git checkout FETCH_HEAD
-                        && python automation/taskcluster/decision_task.py push
+                        && python automation/taskcluster/decision_task.py pull-request
+                    env:
+                      GITHUB_PULL_TITLE: ${pull_request_title}
                   extra:
                     treeherder:
-                      symbol: D
+                      symbol: D-PR
                   metadata:
-                    name: Fenix - Decision task
-                    description: Schedules the build and test tasks for Fenix.
-        else:
-          - $if: 'tasks_for == "cron"'
-            then:
-              $mergeDeep:
-                - {$eval: 'default_task_definition'}
-                - scopes:
-                    - $if: 'trust_level == 3'
-                      then: assume:hook-id:project-mobile/fenix-nightly
-                      else: assume:hook-id:project-mobile/fenix-nightly-staging
-                  routes:
-                    - notify.email.fenix-eng-notifications@mozilla.com.on-failed
-                  payload:
+                    name: 'Fenix - Decision task (Pull Request #${pull_request_number})'
+                    description: 'Building and testing the Fenix - triggered by [#${pull_request_number}](${pull_request_url})'
+        - $if: 'tasks_for == "github-push" && head_branch[:10] != "refs/tags/"'
+          then:
+            $mergeDeep:
+              - {$eval: 'default_task_definition'}
+              - scopes:
+                  - ${assume_scope_prefix}:branch:${short_head_branch}
+                payload:
+                  command:
+                    - >-
+                      git fetch ${repository} ${head_branch}
+                      && git config advice.detachedHead false
+                      && git checkout FETCH_HEAD
+                      && python automation/taskcluster/decision_task.py push
+                extra:
+                  treeherder:
+                    symbol: D
+                metadata:
+                  name: Fenix - Decision task
+                  description: Schedules the build and test tasks for Fenix.
+        - $if: 'tasks_for == "github-release"'
```

Comment:
Note that right now, any github release is assumed to be `beta`.
As we get closer to the Fenix Preview release, I'll add a parse check to see if the tag has `-beta.*` on it